### PR TITLE
add @context support in the rust query planner

### PR DESCRIPTION
**Pull request by clenfest**
Brings Rust query planner to parity with JS.

### CODE REVIEW

**Review by lrlna — COMMENTED** [review]
@goto-bus-stop let's run some perf tests and pay attention to these two things in:
1. rebase
2. edge conditions in compute_nodes_for_op_path_element

File: apollo-federation/src/supergraph/mod.rs [hunk 1]
```diff
@@ -710,6 +713,18 @@ fn extract_object_type_content(
...
+        .and_then(|metadata| metadata.for_identity(&Identity::context_identity()))
+        .and_then(|context_link| CONTEXT_VERSIONS.find(&context_link.url.version))
+        .and_then(|context_spec_def| {
+            context_spec_def
+                .context_directive_name_in_schema(supergraph_schema)
+                .ok()
+                .map(|name_in_schema| (context_spec_def, name_in_schema))
+        });
+    // let context_name_in_schema = context_spec_def
+    //     .and_then(|def| def.context_directive_name_in_schema(supergraph_schema).ok());
```
Comment (lrlna) [inline apollo-federation/src/supergraph/mod.rs:727]:
```suggestion
```

File: apollo-federation/src/query_plan/fetch_dependency_graph.rs [hunk 2]
```diff
@@ -3793,21 +3938,227 @@ fn compute_nodes_for_op_path_element<'a>(
...
+                edge_id,
+                (stack_item.node_id, &stack_item.node_path),
+                stack_item.context,
+                conditions_node_data,
+                created_nodes,
+            )?;
+            updated.node_id = required_node_id;
+            updated.node_path = require_path;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (lrlna) [reply apollo-federation/src/query_plan/fetch_dependency_graph.rs:3986]: we checked with @goto-bus-stop , and this doesn't show up as anything extra on the flamegraphs

**Review by lrlna — APPROVED** [review]
